### PR TITLE
[eglib] Use the XSI-compliant version of strerror_r on apple platforms.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/eglib/gstr.c [hunk 1]
```diff
@@ -46,6 +46,8 @@
 #if (_POSIX_C_SOURCE >= 200112L) && !_GNU_SOURCE
 #define USE_STRERROR_R_XSI
 #endif
+#elif defined(__APPLE__)
```
Comment (cherusker) [inline mono/eglib/gstr.c]: It seems to be not just Apple but probably the whole of Unix ([`strerror_r` at FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=strerror&apropos=0&sektion=3&manpath=FreeBSD+11.1-RELEASE+and+Ports&arch=default&format=html)) ... the macro should probably be defined the other way around: only use the GNU-specific version if on Linux.

**Comment by kumpera** [comment]
Could we probe for this in configure?

**Comment by cherusker** [comment]
Oh, you're absolutely right; [`AC_FUNC_STRERROR_R`](https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Particular-Functions.html) can help to solve this issue!

**Comment by vargaz** [comment]
https://github.com/mono/mono/pull/5453

**Pull request closed without merging**